CMakeLists.txt

```diff
@@ -54,6 +54,12 @@ option(TEST_INTEL_OPTIMIZATION "Test intel optimization and vectorization" OFF)
 option(TEST_ERROR_CHECKING "Enables the error checking unit test. This test will FAIL" OFF)
 option(CODE_COVERAGE "Enable code coverage reporting" OFF)
 
+# HDF5 compression level option, value between 0 and 9
+set(PARTHENON_HDF5_COMPRESSION_LEVEL 5 CACHE STRING "Compression level for HDF5 output and restart files, valid range is 0-9, default is 5, set to 0 to turn off compression")
+if(NOT PARTHENON_HDF5_COMPRESSION_LEVEL MATCHES "^[0-9]$")
+  message(FATAL_ERROR "PARTHENON_HDF5_COMPRESSION_LEVEL must be a number between 0 and 9")
+endif()
+
```

Comment:
Good question. Fortunately, HDF5 handles all the compression and decompression. This flag only specifies with what compression data is written. It can still read data written with any other compression level (or no compression). So if you compile with compression level 5, write a restart dump, then compile with level 0 or anything else, you can still use the previous restart dump. Parthenon never sees the compressed data, because it's all handled by the HDF5 library.